wp-content/plugins/tapestry/endpoints.php

```diff
@@ -6,6 +6,76 @@
 
 require __DIR__ . '/controller/class.tapestry-permissions.php';
 require __DIR__ . '/controller/class.tapestry-controller.php';
+require __DIR__ . '/controller/class.user-controller.php';
+
+ // User endpoints
+
+// Get user's progress
+add_action( 'rest_api_init', function () {
+    register_rest_route( 'tapestry-tool/v1', 'users/progress/', array(
+      'methods' => 'GET',
+      'callback' => 'tapestry_get_user_progress_by_post_id',
+    ) );
+});
+
+// Get user progress on a tapestry page by post id. 
+// Example: http://localhost:8888/tapestry-wp/wp-json/tapestry-tool/v1/users/progress?post_id=44
```

Comment:
Could you move this comment block up to before `add_action` and remove any empty lines between this function declaration and the end of the previous code block? That way, it's all nicely grouped together and easier to read. Do the same for the rest of your endpoints too. Thanks!

Reply:
Reformatted the code in the endpoints file.